src/main/java/com/flightstats/hub/spoke/SpokeReadContentDao.java

```diff
@@ -0,0 +1,167 @@
+
+    @Override
+    public ContentKey insert(String channelName, Content content) throws Exception {
+        throw new NotImplementedException("SpokeReadContentDao.insert not implemented");
+    }
+
+    @Override
+    public SortedSet<ContentKey> insert(BulkContent bulkContent) throws Exception {
+        Traces traces = ActiveTraces.getLocal();
+        traces.add("SpokeReadContentDao.writeBulk");
+        String channelName = bulkContent.getChannel();
+        try {
+            SortedSet<ContentKey> keys = new TreeSet<>();
+            List<Content> items = bulkContent.getItems();
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            ObjectOutputStream stream = new ObjectOutputStream(baos);
+            stream.writeInt(items.size());
+            logger.debug("writing {} items to master {}", items.size(), bulkContent.getMasterKey());
+            for (Content content : items) {
+                content.packageStream();
+                String itemKey = content.getContentKey().get().toUrl();
+                stream.writeInt(itemKey.length());
+                stream.write(itemKey.getBytes());
+                stream.writeInt(content.getData().length);
+                stream.write(content.getData());
+                keys.add(content.getContentKey().get());
+            }
+            stream.flush();
+            traces.add("SpokeReadContentDao.writeBulk marshalled");
+
+            logger.trace("writing items {} to channel {}", items.size(), channelName);
+            if (!spokeStore.newInsert(SpokeStore.READ, channelName, baos.toByteArray(), Cluster.getLocalServer(), ActiveTraces.getLocal(), "bulkKey", channelName)) {
+                throw new FailedWriteException("unable to write bulk to spoke " + channelName);
+            }
+            traces.add("SpokeReadContentDao.writeBulk completed", keys);
+            return keys;
+        } catch (ContentTooLargeException e) {
+            logger.info("content too large for channel " + channelName);
+            throw e;
+        } catch (Exception e) {
+            traces.add("SpokeReadContentDao", "error", e.getMessage());
+            logger.error("unable to write " + channelName, e);
+            throw e;
+        }
+    }
+
+    private String getPath(String channelName, ContentKey key) {
+        return channelName + "/" + key.toUrl();
+    }
+
+    @Override
+    public Content get(String channelName, ContentKey key) {
+        String path = getPath(channelName, key);
+        Traces traces = ActiveTraces.getLocal();
+        traces.add("SpokeReadContentDao.read");
+        try {
+            return spokeStore.newGet(SpokeStore.READ, path, key);
+        } catch (Exception e) {
+            logger.warn("unable to get data: " + path, e);
+            return null;
+        } finally {
+            traces.add("SpokeReadContentDao.read completed");
+        }
+    }
+
+    @Override
+    public SortedSet<ContentKey> queryByTime(TimeQuery query) {
```

Comment:
is any of the query api supported for the read cache?

Reply:
Yeah. I use it actually in testing with the ```location``` query string.